lib/src/theme/headless/switch/switch.theme.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+
+import '../../../../mix.dart';
+
+class SwitchXThemeData {
+  final ThumbThemeData thumb;
+  final TrackThemeData track;
+
+  const SwitchXThemeData.raw({
+    required this.track,
+    required this.thumb,
+  });
+
+  /// The default mix of the switch track.
+  ///
+  /// The default mix is a [Box] widget with a default size of 40x20,
+  /// and a background color of [MaterialTokens.colorScheme.primary],
+  /// and a grey background color when disabled.
+  ///
+  /// The default mix of the switch thumb.
+  ///
+  /// The default mix is a [Box] widget with a default size of 20x20,
+  /// and a background color of [MaterialTokens.colorScheme.onPrimary],
+  /// and a grey background color when disabled.
+  ///
+  SwitchXThemeData({
+    ThumbThemeData? thumb,
+    TrackThemeData? track,
+  }) : this.raw(
+          thumb: thumb ?? ThumbThemeData(),
+          track: track ?? TrackThemeData(),
+        );
+}
+
+class SwitchXThemeProp {
+  /// The size of the track
+  final Size size;
+
+  /// The color of the track
+  final Color color;
+
+  /// The color of the track when disabled
+  final Color disabledColor;
+
+  const SwitchXThemeProp({
```

Comment:
Why don't use just a mix prop here ? 
Instead of apply some opinions in what props the SwitchXThemeProps should have..